nullaway/src/main/java/com/uber/nullaway/handlers/AbstractFieldContractHandler.java

```diff
@@ -163,7 +163,31 @@ protected boolean validateAnnotationSyntax(
                   null));
       return false;
     } else {
+      Symbol.ClassSymbol classSymbol =
+          castToNonNull(ASTHelpers.enclosingClass(methodAnalysisContext.methodSymbol()));
       for (String fieldName : content) {
+        if (isStaticThisAnnotationField(classSymbol, fieldName)) {
+          message =
+              "Currently, @"
+                  + annotName
+                  + " does not support this. annotations for static fields. ";
```

Comment:
The message can just be `"Cannot reference to static field " + fieldName + " using this"`

Reply:
Fixed!